element-connector/src/main/java/org/eclipse/californium/elements/TcpCorrelationContext.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - add support for correlation context to provide
+ *                                      additional information to application layer for
+ *                                      matching messages (fix GitHub issue #1)
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+/**
+ * A correlation context that explicitly supports TCP specific properties.
+ */
+public class TcpCorrelationContext extends MapBasedCorrelationContext {
+
+	public static final String KEY_CONNECTION_ID = "CONNECTION_ID";
```

Comment:
See NettyContextUtils. But the "epoch" seems to be not available. 
I'm currently working on the finalize of this work. I changed the comment

> /**
>  * A correlation context that explicitly supports TLS specific properties.
>  * Currently the context is not aware of renegotiation (API to acquire
>  * information is missing). According oracle, the renegotiate issues seems not
>  * be fixed, if your java is not to deprecated.
>  * 
>  * @see http://www.oracle.com/technetwork/java/javase/overview/tlsreadme2-176330.html
>  */
> public class TlsCorrelationContext extends TcpCorrelationContext {
> 

to indicate the information behind that.